### ISSUE TITLE
Switch MultiTypeSymbolTable from associative domain/array to map (?)

### ISSUE THREAD

**Issue by bradcray** [mhmerrill/arkouda#169]
This issue captures a notion to replace the use of an associative domain and array pair in MultiTypeSymbolTable.chpl to a map (new in Chapel 1.20's standard library) since that's the basic pattern being used.  This seems like it would simplify the code and also support a non-nilable element type; however, in trying to do so today, I'm running into fragilities in the map type when the valueType is a managed class type, such as:

* https://github.com/chapel-lang/chapel/issues/14370
* https://github.com/chapel-lang/chapel/issues/14361

**Comment by bradcray** [comment]
My start at this is here:

https://github.com/bradcray/arkouda/tree/arkouda-nilable-with-map

However, I'm running into lifetime issues in trying to use the map type, as reported in https://github.com/chapel-lang/chapel/issues/14370

I think switching from `shared` to `owned` (issue #170) could help with the lifetime issues, but then would run into other fragile interactions between owned and nilability (e.g., https://github.com/chapel-lang/chapel/issues/14361)

**Comment by jwcrandall** [comment]
@bradcray Is it possible to work on #176 while you are working on #169 ?

**Comment by bradcray** [comment]
Yeah, that was my hope—to tackle both at once.  I'm not actively working on either of these at the present, though.

**Comment by jwcrandall** [comment]
Ok, I going to try to understand #176 .

**Comment by bradcray** [comment]
Ah, I think I misunderstood and was asking whether I could work on #176 while doing #169 (but re-reading, I interpret it as you being interested in working on it).

I think it would be challenging to crack #176 until Chapel's maps improve further.  For example, [this issue](https://github.com/chapel-lang/chapel/issues/14861) is currently actively being worked on the Chapel side, and my understanding is that until it is resolved, maps of non-nilable classes have substantial problems.

**Comment by jwcrandall** [comment]
Ok, I will follow the progress of https://github.com/chapel-lang/chapel/issues/14861
